pkg/ccl/streamingccl/streamproducer/stream_event_batcher.go

```diff
@@ -49,6 +52,41 @@ func (seb *streamEventBatcher) addDelRange(d *kvpb.RangeFeedDeleteRange) {
 	seb.size += d.Size()
 }
 
+// addSpanConfigs adds a slice of spanConfig entries that were recently flushed
+// by the rangefeed cache. The function elides duplicate updates by checking
+// that an update is newer than the tracked frontier and by checking for
+// duplicates within the inputEvents. This function assumes that the input events are
+// timestamp ordered and that the largest timestamp in the batch is the rangefeed frontier.
+func (seb *streamEventBatcher) addSpanConfigs(inputEvents []streampb.StreamedSpanConfigEntry) {
+
+	// eventSeenAtCurrentTimestamp is used to track unique events within the
+	// inputEvents at the current timestamp.
+	eventsSeenAtCurrentTimestamp := make(map[string]struct{})
+	var currentTimestamp hlc.Timestamp
+
+	for _, event := range inputEvents {
+		if event.Timestamp.LessEq(seb.spanConfigFrontier) {
+			// Any event with a timestamp at or below the rangefeed frontier is a duplicate.
+			continue
+		}
+		if currentTimestamp.Less(event.Timestamp) {
+			for key := range eventsSeenAtCurrentTimestamp {
+				delete(eventsSeenAtCurrentTimestamp, key)
+			}
+			currentTimestamp = event.Timestamp
+		}
+		stringedEvent := event.String()
```

Comment:
[potential-followup] The processing of these events seems expensive. We iterate through all the updates to create StreamedSpanConfigEntries and then iterate through the StreamedSpanConfigEntries in this loop and marshal the protobuf into text.  I kinda wonder if we could do some of this as we go in the original loop and whether there is a smaller unique key we can use.

But, I'm OK with it for now as we try to get things wired up.